extensions/stats/plugin.h

```diff
@@ -307,45 +325,51 @@ class PluginRootContextInbound : public PluginRootContext {
 // Per-stream context.
 class PluginContext : public Context {
  public:
-  explicit PluginContext(uint32_t id, RootContext* root)
-      : Context(id, root), is_tcp_(false) {
-    request_info_ = std::make_shared<::Wasm::Common::RequestInfo>();
-  }
+  explicit PluginContext(uint32_t id, RootContext* root) : Context(id, root) {}
 
+  // Called for both HTTP and TCP streams, as a final data callback.
   void onLog() override {
-    if (!rootContext()->initialized()) {
-      return;
-    }
-    if (is_tcp_) {
-      cleanupTCPOnClose();
+    rootContext()->deleteFromRequestQueue(id());
+    if (request_info_.request_protocol == ::Wasm::Common::Protocol::TCP) {
+      request_info_.tcp_connections_closed++;
     }
-    rootContext()->report(*request_info_, is_tcp_);
+    rootContext()->report(request_info_, true);
   };
 
-  FilterStatus onNewConnection() override {
-    if (!rootContext()->initialized()) {
-      return FilterStatus::Continue;
+  // HTTP streams start with headers.
+  FilterHeadersStatus onRequestHeaders(uint32_t, bool) override {
+    ::Wasm::Common::populateRequestProtocol(&request_info_);
+    // Save host value for recurrent reporting
+    if (rootContext()->useHostHeaderFallback()) {
```

Comment:
It is not clear to me why we fetch request host here? Also since we are fill in url host header here, we should remove it from `populateExtendedHTTPRequestInfo`.

Reply:
It's filled here because onTick will happen outside onRequestHeaders, and url_host is only available at either onRequestHeader or onLog. URL host is needed for metadata fallback.

I didn't remove it from extended info because StackDriver does not set it separately (same cause as the one @gargnupur pointed out for TCP).

Reply:
Ah I see, I think `url_host is only available at either onRequestHeader or onLog` worths to mentioning in the comment.

Reply:
Done. TBH there is no reason why it shouldn't be available. Just ABI issue.